src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -285,9 +316,33 @@ public synchronized IndexShard createShard(int sShardId, boolean primary) {
         Injector shardInjector = null;
         try {
 
+            
+
             ShardPath path = ShardPath.loadShardPath(logger, nodeEnv, shardId, indexSettings);
             if (path == null) {
-                path = ShardPath.selectNewPathForShard(nodeEnv, shardId, indexSettings);
+
+                long totFreeSpace = 0;
```

Comment:
It's a little tricky because I'm iterating over IndexService's shards map ... but maybe I could break that out into a loop that first produces a map of shard to data path, and then pass that map to a new static method in ShardPath.  I'll try.